app/org/sagebionetworks/bridge/dynamodb/DynamoSchedulePlanDao.java

```diff
@@ -108,7 +108,11 @@ public SchedulePlan createSchedulePlan(StudyIdentifier studyIdentifier, Schedule
     public SchedulePlan updateSchedulePlan(StudyIdentifier studyIdentifier, SchedulePlan plan) {
         checkNotNull(studyIdentifier);
         checkNotNull(plan);
-        
+
+        SchedulePlan saved = getSchedulePlan(studyIdentifier, plan.getGuid());
+        if (saved == null || (saved.isDeleted() && plan.isDeleted())) {
+            throw new EntityNotFoundException(SchedulePlan.class);
+        }
```

Comment:
Unit test for updating a missing schedule plan?

Reply:
Added two tests to test EntityNotFoundException is thrown for missing schedule plan and for deleted schedule plan when the provided update is also deleted.